### ISSUE TITLE
Help with unresolved file paths during plugin execution for banDuplicateClasses

### ISSUE THREAD

**Issue by sudr** [mojohaus/extra-enforcer-rules#29]
It appears that there is a bug when using the banDuplicateClasses rule where the file path for the dependency that is being checked hasn't been resolved during the plugin execution.

I've created a minimal pom that illustrates this issue: https://github.com/sudr/test-dup-classes

When I execute this via: 
`mvn org.apache.maven.plugins:maven-enforcer-plugin:enforce`

I get the following in the logs:

```
[INFO] --- maven-enforcer-plugin:1.4.1:enforce (default-cli) @ test-dup-classes ---
[WARNING] Could not find org.hamcrest:hamcrest-core:jar:1.3:compile at null

```

**Comment by mfriedenhagen** [comment]
hello, this is maybe a documentation bug. With resolution dependencies are not known, so when using  this rule you need at.least have to reach the compile phase. This is hoe Maven works :-)

**Comment by sudr** [comment]
You are right. If I execute this as:

`mvn install org.apache.maven.plugins:maven-enforcer-plugin:enforce`

it works like you explained.

What I found was that even when invoking this with the install goal, if the maven project packaging is "pom" (this is a parent pom project) then the top level dependencies declared in the pom are not resolved and log this warning. But the transitive dependencies are being resolved and the execution fails as expected when there are duplicates.

So I'm thinking this plugin should only be used with non-pom packaging and should be invoked with the install goal.

**Comment by mfriedenhagen** [comment]
Hello, your pom project should only manage dependencies, it depends per se on nothing :-).  Because the compile plugin is not invoked for pom projects, there is probably no compile, test-compile or runtime classpath.
